### PR TITLE
♻️  Rename HttpPushConsumer to DestinationConsumer

### PR DESCRIPTION
**Pull request by RTLS**
In preparation for making this struct polymorphic and supporting other
destinations

Co-authored-by: Anthony Accomazzo <accomazz@gmail.com>

### CODE REVIEW

**Comment by RTLS** [comment]
* **#538** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/538?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RTLS and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/sequin/consumers/consumers.ex [hunk 1]
```diff
@@ -1374,7 +1374,7 @@ defmodule Sequin.Consumers do
   def delete_http_endpoint(%HttpEndpoint{} = http_endpoint) do
     http_endpoint
     |> Ecto.Changeset.change()
-    |> Ecto.Changeset.foreign_key_constraint(:http_push_consumers, name: "http_push_consumers_http_endpoint_id_fkey")
+    |> Ecto.Changeset.foreign_key_constraint(:destination_consumers, name: "destination_consumers_http_endpoint_id_fkey")
```
Comment (acco) [inline lib/sequin/consumers/consumers.ex:1377]: todo: migrations

File: lib/sequin/health/health.ex [hunk 2]
```diff
@@ -34,7 +34,7 @@ defmodule Sequin.Health do
     field :entity_kind,
           :http_endpoint
           | :http_pull_consumer
-          | :http_push_consumer
+          | :push_consumer
```
Comment (acco) [inline lib/sequin/health/health.ex:37]: revert

File: lib/sequin_web/live/consumers/form.ex [hunk 3]
```diff
@@ -397,14 +397,14 @@ defmodule SequinWeb.ConsumersLive.Form do
...
-          Consumers.create_http_push_consumer_for_account_with_lifecycle(account_id, params)
+        %DestinationConsumer{} ->
+          Consumers.create_destination_consumer_for_account_with_lifecycle(account_id, params)
       end
 
     case case_result do
       {:ok, consumer} ->
         kind = Consumers.kind(consumer)
-        consumer_type = if kind == :pull, do: "HttpPullConsumer", else: "HttpPushConsumer"
+        consumer_type = if kind == :pull, do: "HttpPullConsumer", else: "DestinationConsumer"
```
Comment (acco) [inline lib/sequin_web/live/consumers/form.ex:407]: revert

**Pull request merged**